src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy

```diff
@@ -0,0 +1,72 @@
+package life.qbic.samplestatus.reporter.commands
+
+import life.qbic.samplestatus.reporter.Result
+import life.qbic.samplestatus.reporter.SampleStatusReporter
+import life.qbic.samplestatus.reporter.SampleUpdate
+import life.qbic.samplestatus.reporter.api.LimsQueryService
+import life.qbic.samplestatus.reporter.api.UpdateSearchService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import picocli.CommandLine
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+/**
+ * <b>Command that runs the app for a given instant</b>
+ *
+ * @since 1.0.0
+ */
+@CommandLine.Command(name = "SampleStatusReporter", version = "1.0.0", mixinStandardHelpOptions = true)
+class ReportSinceInstant implements Runnable {
+
+  private static final Logger log = LoggerFactory.getLogger(ReportSinceInstant.class)
+
+  @CommandLine.Option(names = ["-t", "--time-point"], description = "Point in time from where to search for updates e.g. '2022-01-01T00:00:00Z'.\nDefaults to the last successful run. \nIf never run successfully defaults to the same time yesterday.")
+  final Instant timePoint
+
+  private final LimsQueryService limsQueryService
+  private final SampleStatusReporter statusReporter
+  private final UpdateSearchService updateSearchService
+
+  ReportSinceInstant(LimsQueryService limsQueryService, SampleStatusReporter statusReporter, UpdateSearchService updateSearchService) {
+    this.limsQueryService = limsQueryService
+    this.statusReporter = statusReporter
+    this.updateSearchService = updateSearchService
+    if (!timePoint) {
+      timePoint = defaultTimePoint
+    }
+  }
+
+  private Instant defaultTimePoint = updateSearchService.getLastUpdateSearchTimePoint().orElse(Instant.now().minus(1, ChronoUnit.DAYS))
```

Comment:
does this work? can you guarantee that this is executed after the updateSearchService is available to the instance?

Reply:
Yes. I tested it locally and this is called from the constructor so the updateSearchService is available at that time 😄.
As `defaultTimePoint` is no variable or method but is evaluated dynamically this is only executed when called. And as it is only called after the `updateSearchService` is instantiated this works.

Reply:
It changed it as I am unsure how the order of execution is here and whether it is always the same